core/trino-main/src/main/java/io/trino/operator/DirectExchangeClientFactory.java

```diff
@@ -123,7 +125,12 @@ public ThreadPoolExecutorMBean getExecutor()
     }
 
     @Override
-    public DirectExchangeClient get(LocalMemoryContext systemMemoryContext, TaskFailureListener taskFailureListener, RetryPolicy retryPolicy)
+    public DirectExchangeClient get(
+            QueryId queryId,
+            ExchangeId exchangeId,
```

Comment:
change of this interface should go to last commit. New args are not used in `Identify exchanges by ExchangeId`